### PR TITLE
chore: Clean up CI

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

This aligns the SDK CI with other repositories

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: .github/workflows/lint_markdown.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       - name: Vale
         uses: errata-ai/vale-action@v2
         with:
-          vale_flags: "--glob='!{plugins/.snapshots/*,CHANGELOG.md,.github/styles/proselint/README.md}'"
+          vale_flags: "--glob=!{plugins/.snapshots/*,CHANGELOG.md,.github/styles/proselint/README.md}"
```
Comment (erezrokah) [inline .github/workflows/lint_markdown.yml:18]: Vale doesn't work with the `''` quotes

File: .github/workflows/release-pr.yml [hunk 2]
```diff
@@ -15,8 +15,6 @@ jobs:
           package-name: plugin-sdk
           token: ${{ secrets.GH_CQ_BOT }}
           pull-request-title-pattern: "chore${scope}: Release${component} v${version}"
-          bump-minor-pre-major: true
```
Comment (erezrokah) [inline .github/workflows/release-pr.yml:18]: We don't need this anymore as we released our major version

File: .github/workflows/lint_markdown.yml (hunk 1 above)
Comment (candiduslynx) [inline .github/workflows/lint_markdown.yml:18]: Shouldn't it be
```suggestion
          vale_flags: "--glob={plugins/.snapshots/*,CHANGELOG.md,.github/styles/proselint/README.md}"
```

Comment (erezrokah) [reply .github/workflows/lint_markdown.yml:18]: Good question, `--glob` says which files to lint, so if we want to ignore files we need the `!`

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**